plugins/admin-settings/general-settings/index.jsx

```diff
@@ -0,0 +1,161 @@
+import { __ } from '@wordpress/i18n';
+import { Button, TextControl, withNotices } from '@wordpress/components';
+import ImagePicker from '@/components/imagePicker';
+import { Sortable, SortableItem, useOption } from '@alleyinteractive/block-editor-tools';
+
+const AdminGeneralSettings = withNotices(
+  ({ noticeOperations, noticeUI }) => {
+    const {
+      value: settings = {},
+      isEdited = false,
+      isSaving = false,
+      onChange = () => {},
+      onSave = () => {},
+    } = useOption('nb_settings');
+
+    const {
+      from_email: fromEmail = '',
+      reply_to_email: replyToEmail = '',
+      from_names: fromNames = [],
+      facebook_url: facebookUrl = '',
+      twitter_url: twitterUrl = '',
+      instagram_url: instagramUrl = '',
+      youtube_url: youtubeUrl = '',
+      image = 0,
+      address = '',
+    } = settings;
+
+    /**
+     * Update settings array with new data.
+     * @param {string} key Key to update.
+     * @param {string} value Value to update.
+     */
+    const updateSettingsData = (key, value) => {
```

Comment:
By creating our own custom hook we could also encapsulate this functionality so that the implementing component gets a getter and a setter for each value, for instance.